### ISSUE TITLE
Add a warning when importing file with wrong casing on a case insensitive file system

### ISSUE THREAD

**Issue by abique** [slint-ui/slint#4265]
Hi,

It'd be great to be able to make slint build case sensitive.

By default issue a warning if there is a case mismatch.
Whith a cargo feature, turn case mismatch into an error.

This will help to catch errors earlier when working on macOS/Windows and then building on Linux later on.

Thanks,
Alex

**Comment by ogoffart** [comment]
Do I understand correctly that you want to catch error such as:

```
import { Foo } from "FooBar.slint";
```

where the file on the disk is actually `foobar.slint` (all lower case).

This would not be an error on Windows since the file system is case insensitive, but would be an error on Linux.

I think we indeed should make this a warning on Windows.

**Comment by abique** [comment]
Yes, you've understood correctly.
I think it should be a warning by default, and it should be possible to turn it into an error with some flags maybe in cargo.

**Comment by ogoffart** [comment]
This shouldn't be to hard to implement.

It should probably be done in ensure_document_loaded https://github.com/slint-ui/slint/blob/424304045331d5ef7881befbab3ca540bc9db637/internal/compiler/typeloader.rs#L285

If the last component of `path_cannon` and `file_to_import`  differ only in casing, we should emit a warning (using `borrowed_state.diag.push_warning` as already done in that function)
